ethereum/contracts/IncentivizedReceiveChannel.sol

```diff
@@ -0,0 +1,173 @@
+pragma solidity >=0.6.2;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./Decoder.sol";
+
+contract IncentivizedReceiveChannel {
+    using Decoder for bytes;
+
+    uint256 public lastProcessedNonce;
+
+    uint256 public MAX_PAYLOAD_BYTE_SIZE = 1000;
+    uint256 public MAX_PAYLOAD_GAS_COST = 500000;
+    uint256 public EXTERNAL_CALL_COST = 21000;
+    uint256 public MAX_GAS_PER_MESSAGE =
+        EXTERNAL_CALL_COST + MAX_PAYLOAD_BYTE_SIZE + EXTERNAL_CALL_COST;
+
+    struct Message {
+        uint256 nonce;
+        string senderApplicationId;
+        address targetApplicationAddress;
+        bytes payload;
+    }
+
+    struct Commitment {
+        bytes commitmentHash;
+    }
+
+    struct CommitmentContents {
+        Message[] messages;
+    }
+
+    constructor() public {
+        lastProcessedNonce = 0;
+    }
+
+    event MessageDelivered(uint256 _nonce, bool _result);
+
+    function newParachainCommitment(
+        Commitment memory commitment,
+        CommitmentContents memory commitmentContents,
+        uint256 parachainBlockNumber,
```

Comment:
This is not used – do we really need it?

Reply:
i feel like i put this here for a reason, though can't remember now lol, will think about it and see